starter-core/src/main/java/io/micronaut/starter/feature/lang/groovy/application.rocker.raw

```diff
@@ -34,11 +35,19 @@ class Application {
     static class DefaultEnvironmentConfigurer implements ApplicationContextConfigurer {
         @@Override
         public void configure(@@NonNull ApplicationContextBuilder builder) {
-            builder.defaultEnvironments("@(defaultEnvironment)");
+            builder.defaultEnvironments("@(defaultEnvironment)")
+@if (features.contains(Crac.NAME)) {
```

Comment:
I would prefer if we pass a boolean with `eagerInitSingleton` and remove features logic from rocker templates

Reply:
I created an interface `RequireEagerSingletonInitializationFeature` which extends `Feature` and switched `Crac` to implement that instead

We then check for any `RequireEagerSingletonInitializationFeature` features, and pass a boolean to the template